### PR TITLE
CVSL-2260: Add endpoint to get residential checks for an assessment.

### PR DESCRIPTION
**Pull request by justin-gardiner**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/ResidentialChecksService.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  private val assessmentService: AssessmentService,
+) {
+  fun getResidentialChecksView(prisonNumber: String): ResidentialChecksView {
+    val currentAssessment = assessmentService.getCurrentAssessmentSummary(prisonNumber)
+
+    val tasks = RESIDENTIAL_CHECKS_POLICY_V1.tasks.map { task ->
+      ResidentialChecksTaskProgress(
+        code = task.code,
+        taskName = task.name,
+        status = ResidentialChecksTaskStatus.NOT_STARTED,
```
Comment (justin-gardiner) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/ResidentialChecksService.kt:21]: All status will be NOT_STARTED for now as we have no way of persisting / updating status of tasks yet.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/ResidentialChecksResource.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.model.ResidentialChecksView
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.service.ResidentialChecksService
+
+@RestController
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class ResidentialChecksResource(private val residentialChecksService: ResidentialChecksService) {
+  @GetMapping("/offender/{prisonNumber}/current-assessment/residential-checks")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/ResidentialChecksResource.kt:23]: would this need to be for a specific curfew address check?

Comment (justin-gardiner) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/ResidentialChecksResource.kt]: fixed

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**